src/btree/bt_cursor.c

```diff
@@ -2201,26 +1902,16 @@ __wt_btcur_range_truncate(WT_TRUNCATE_INFO *trunc_info)
     if (logging)
         WT_RET(__wt_txn_truncate_log(trunc_info));
 
-    switch (btree->type) {
-    case BTREE_COL_FIX:
-        WT_ERR(__cursor_truncate_fix(start, stop, __cursor_col_modify));
-        break;
-    case BTREE_COL_VAR:
-        WT_ERR(__wt_cursor_truncate(start, stop, __cursor_col_modify));
-        break;
-    case BTREE_ROW:
-        /*
-         * The underlying cursor comparison routine requires cursors be fully instantiated when
-         * truncating row-store objects because it's comparing page and/or skiplist positions, not
-         * keys. (Key comparison would work, it's only that a key comparison would be relatively
-         * expensive, especially with custom collators. Column-store objects have record number
-         * keys, so the key comparison is cheap.) The session truncate code did cursor searches when
-         * setting up the truncate so we're good to go: if that ever changes, we'd need to do
-         * something here to ensure a fully instantiated cursor.
-         */
-        WT_ERR(__wt_cursor_truncate(start, stop, __cursor_row_modify));
-        break;
-    }
+    /*
+     * The underlying cursor comparison routine requires cursors be fully instantiated when
+     * truncating row-store objects because it's comparing page and/or skiplist positions, not keys.
+     * (Key comparison would work, it's only that a key comparison would be relatively expensive,
+     * especially with custom collators. Column-store objects have record number keys, so the key
+     * comparison is cheap.) The session truncate code did cursor searches when setting up the
+     * truncate so we're good to go: if that ever changes, we'd need to do something here to ensure
+     * a fully instantiated cursor.
+     */
+    WT_ERR(__wt_cursor_truncate(start, stop, __cursor_modify));
```

Comment:
So while this cleanup is nice, there is the teeny-tiniest performance concern here as now we've buried an if statement in the truncate loop. I don't think it'll show up in perf testing.